src/lib/components/FornaContainer.react.js

```diff
@@ -0,0 +1,274 @@
+
+        this._fornaContainer = new PreFornaContainer('#' + id, {
+            initialSize: [width, height],
+        });
+        this.renderNewSequences();
+    }
+
+    renderNewSequences() {
+        const {sequences} = this.props;
+
+        if (this._fornaContainer) {
+            this._fornaContainer.clearNodes();
+
+            sequences.forEach(seq => {
+                const unpackedOptions = Object.assign({}, seq.options, {
+                    sequence: seq.sequence,
+                    structure: seq.structure,
+                });
+                this._fornaContainer.addRNA(seq.structure, unpackedOptions);
+            });
+        }
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const {sequences, colorScheme, nodeFillColor} = this.props;
+
+        if (Boolean(sequences) !== Boolean(nextProps.sequences)) {
+            return true;
+        }
+
+        if (sequences.length !== nextProps.sequences.length) {
+            return true;
+        }
+
+        const sequence_compare = R.zip(sequences, nextProps.sequences);
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    seq[0].sequence !== seq[1].sequence ||
+                    seq[0].structure !== seq[1].structure
+            )
+        ) {
+            return true;
+        }
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    Boolean(seq[0].options) !== Boolean(seq[1].options) ||
+                    (Boolean(seq[0].options) &&
+                        Boolean(seq[1].options) &&
+                        R.symmetricDifference(
+                            Object.keys(seq[0].options),
+                            Object.keys(seq[1].options)
+                        ).length > 0)
+            )
+        ) {
+            return true;
+        }
+
+        if (
+            sequence_compare.some(
+                seq =>
+                    Boolean(seq[0].options) &&
+                    Boolean(seq[1].options) &&
+                    Object.keys(seq[0].options).some(
+                        _option =>
+                            seq[0].options._option !== seq[1].options._option
+                    )
+            )
+        ) {
+            return true;
+        }
+
+        this._fornaContainer.addCustomColors(nextProps.customColors);
+
+        if (nodeFillColor !== nextProps.nodeFillColor) {
+            this._fornaContainer.setOutlineColor(nextProps.nodeFillColor);
+        }
+
+        this._fornaContainer.changeColorScheme(colorScheme);
+
+        return false;
+    }
+
+    render() {
+        this.renderNewSequences();
+        return <div id={this.props.id} style={{outline: 'none'}} />;
+    }
+}
+
+FornaContainer.propTypes = {
+    /**
+     * The ID of this component, used to identify dash components in
+     * callbacks. The ID needs to be unique across all of the
+     * components in an app.
+     */
+    id: PropTypes.string.isRequired,
```

Comment:
If I'm reading this right, `FornaContainer` can accept either a selector string or a DOM node (it's just getting [passed to `d3.select`](https://github.com/ViennaRNA/fornac/blob/master/app/scripts/fornac.js#L802)). If you use a `ref` and pass that instead of `'#' + id`, then id becomes optional, making it easier to use and more robust.